File: enheng/app/src/main/java/com/yuge/enheng/MyViewModel.java
package com.yuge.enheng;

import androidx.lifecycle.MutableLiveData;

public class MyViewModel extends androidx.lifecycle.ViewModel {
    private MutableLiveData<Integer> likedNumber;


    public MutableLiveData<Integer> getLikedNumber() {
        if(likedNumber==null){
            likedNumber=new MutableLiveData<>();
            likedNumber.setValue(0);
        }
        return likedNumber;
    }

    public void addLikedNumber(int n){
        likedNumber.setValue(likedNumber.getValue()+n);
    }
}

File: kotlinDemo/app/src/main/java/com/yuge/kotlindemo/MyViewModel.kt
package com.yuge.kotlindemo

import androidx.lifecycle.MutableLiveData
import androidx.lifecycle.ViewModel

class MyViewModel:ViewModel() {
    var number:MutableLiveData<Int> by lazy  { MutableLiveData<Int>().also { it.value=0 } }

//    init {
//        number=MutableLiveData()
//        number.value=0
//    }
}